telegrinder/bot/rules/text.py

```diff
@@ -10,14 +11,28 @@ class TextMessageRule(MessageRule, ABC, require=[HasText()]):
     pass
 
 
-class Text(TextMessageRule):
+class Text(TextMessageRule, TranslatableRuleMixin):
     def __init__(self, texts: str | list[str], ignore_case: bool = False):
         if not isinstance(texts, list):
             texts = [texts]
         self.texts = texts if not ignore_case else list(map(str.lower, texts))
         self.ignore_case = ignore_case
 
-    async def check(self, message: Message, ctx: dict) -> bool:
+    def check_text(self, text: str, override_texts: list[str] | None = None) -> bool:
+        texts = override_texts or self.texts
         return (
-            message.text if not self.ignore_case else message.text.lower()
-        ) in self.texts
+            text if not self.ignore_case else text.lower()
+        ) in texts
+
+    async def check(self, message: Message, ctx: dict) -> bool:
+        return self.check_text(message.text)
+
+    def translate(self) -> "TranslatedText":
+        return TranslatedText(inner_rule=self)
+
+
+class TranslatedText(ABCTranslatedRule[Text]):
```

Comment:
nonsense it is better to just generate a `Text` rule with translated Text.texts

Reply:
You've probably misunderstood my idea of translated rules. `translate()` doesn't translate texts to specified locale, instead it creates a new rule that takes translator context and translates texts inside the `check()` method. We've discussed more in telegram dm.

Reply:
I will redesign it soon as for now it looks just bad